CHANGELOG.md

```diff
@@ -106,6 +106,9 @@
 - Apply Konnect-related customizations to `DataPlane`s that properly reference `KonnectExtension`
   resources.
   [#714](https://github.com/Kong/gateway-operator/pull/714)
+- The KonnectExtension functionality is enabled only when the `--enable-controller-konnect`
+  flag is set.
```

Comment:
Might also mention the env `GATEWAY_OPERATOR_ENABLE_CONTROLLER_KONNECT` which is equivalent.